src/hermezjs-typings.d.ts

```diff
@@ -703,19 +697,35 @@ declare module "@hermeznetwork/hermezjs/src/enums" {
     Withdraw = "Withdrawn",
     Exit = "Exit",
     ForceExit = "ForceExit",
-  };
-
-  declare const TxState = {
+  } as const;
+
+  export type TxTypesUnion =
```

Comment:
Precisely because `enums` are both types and values, they can not be used in TypeScript Declaration files (`.d.ts`).

I've learnt this in this PR when I realized I was having problems in runtime when importing the enums to use them as values. They are never available as values at runtime. So I switched to [const assertions](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#const-assertions) so we can keep using these definitions as both values and types (using `typeof`).

I also decided to export the unions `TxTypesUnion` and `TxStatesUnion` because they are used in several places.

Some more info on the topic: https://lukasbehal.com/2017-05-22-enums-in-declaration-files/

Reply:
The problems I had were:

  * Using string literals as values for enum types
  * Naming the enums differently than the consts exported by the lib (TxType and TxState)

Fixed!